src/ape/managers/chain.py

```diff
@@ -359,9 +360,23 @@ def base_fee(self) -> int:
     def pending_timestamp(self) -> int:
         """
         The current epoch time of the chain, as an ``int``.
+        Set the timestamp to fast-forward time into the future.
+
+        Usage example::
+
+            from ape import chain
+
+            chain.pending_timestamp += 3600
         """
 
-        return int(time.time())
+        pending_block = self.provider.get_block("pending")
+        return int(pending_block.timestamp + self._time_offset)
+
+    @pending_timestamp.setter
+    def pending_timestamp(self, new_value: int):
+        provider = self._get_test_provider(TestProviderAPI.set_timestamp.__name__)
+        new_timestamp = provider.set_timestamp(new_value)
+        self._time_offset = new_timestamp - self.pending_timestamp
```

Comment:
Shouldn't actually need to track this anymore now that you are using `self.provider.get_block("pending").timestamp`!